abis_mapping/plugins/related_site_id_part_of_lookup.py

```diff
@@ -0,0 +1,45 @@
+"""Module providing custom relatedSiteID lookup checks when value of relationshipToRelatedSite is partOf"""
+
+# Third-party
+import attrs
+import frictionless
+import frictionless.checks
+import frictionless.errors
+
+# Typing
+from typing import Iterable
+
+
+@attrs.define(kw_only=True, repr=False)
+class RelatedSiteIDPartOfLookup(frictionless.Check):
+    """Specific check confirming relationshipToRelatedSite is a valid siteID when relationship is `partOf`.
+
+    Attributes:
+        site_ids: All siteIDs provided in the template.
+    """
+
+    Errors = [frictionless.errors.RowConstraintError]
+    site_ids: set[str]
```

Comment:
FYI using the built-in `set` as a type indicates a mutable set.
You can use `collections.abc.Set` as an alternative which is the abstract type for a read-only set.

If you changed this you can still pass a normal `set` when instantiating this class,
the only change is mypy would error if we accidentally mutate that set in this class, which would be nice.

Reply:
its a mutable set. Doesn't matter to the implementation.

Reply:
it was intended as a mutable set.